graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/runtime/JSConfig.java

```diff
@@ -130,8 +130,8 @@ public final class JSConfig {
     public static final boolean FastOwnKeys = true;
     /** AST-level inlining of trivial built-in functions (e.g. String.prototype.charAt). */
     public static final boolean InlineTrivialBuiltins = true;
-    /** [Construct] as part of the CallTarget names. Off by default (footprint). */
-    public static final boolean DetailedCallTargetNames = false;
```

Comment:
Considering that this flag has some footprint impact, it would be nice to keep the possibility to turn it off. How about basing the value of this flag on a system property (similar to `DebugCounters` flag)?

Reply:
Sounds like a good compromise for me. Thanks!

Reply:
Do you plan to update this PR accordingly? I can do the change myself, but if you want to keep a commit authored by you in `graal-js` history then it must be done here in this PR ;-).

Reply:
I just pushed a patch with your suggestion.

Reply:
Looks good, thanks! But could you, please, get rid of the first commit (i.e. squash the commits)? I will merge the PR then. Thank you in advance.